.travis.yml

```diff
@@ -28,7 +29,10 @@ rust: &rust
     - cd zeromq-4.2.5 && ./configure --prefix=$HOME --with-libsodium && make && make install && cd ..
 
   script:
-  - cargo clean
+  - |
+    if [[ "$TRAVIS_BRANCH" == staging ] || [ "$TRAVIS_BRANCH" == trying ] || [ "$TRAVIS_PULL_REQUEST_BRANCH" == staging ] || [ "$TRAVIS_PULL_REQUEST_BRANCH" == trying ]]; then
+      cargo clean
```

Comment:
Can we remove `cargo clean` from all branches? Is there any particular reason to do `cargo clean`?

Reply:
It's for reproducibility -- especially for the "git" repositories, if they are force-pushed / rebased, the originally dependency may not work... but one would only detect that if it was built from scratch